bridge/svix-bridge-plugin-kafka/src/input.rs

```diff
@@ -0,0 +1,214 @@
+                .map_err(Error::Deserialization)?
+        } else {
+            serde_json::from_slice(payload).map_err(Error::Deserialization)?
+        };
+
+        let CreateMessageRequest {
+            app_id,
+            message,
+            post_options,
+        } = payload;
+
+        // FIXME(jplatte): I think we should always set the idempotency key
+
+        self.svix_client
+            .message()
+            .create(app_id, message, post_options.map(Into::into))
+            .await?;
+
+        Ok(())
+    }
+
+    async fn transform(&self, script: String, input: TransformerInput) -> Result<JsObject> {
+        let (job, rx) = TransformerJob::new(script, input);
+        self.transformer_tx
+            .as_ref()
+            .ok_or_else(|| Error::transformation("transformations not configured"))?
+            .send(job)
+            .map_err(|e| Error::transformation(e.to_string()))?;
+
+        let ret = rx
+            .await
+            .map_err(|_e| Error::transformation("transformation rx failed"))
+            .and_then(|x| {
+                x.map_err(|_e| Error::transformation("transformation execution failed"))
+            })?;
+
+        match ret {
+            TransformerOutput::Object(v) => Ok(v),
+            TransformerOutput::Invalid => Err(Error::transformation(
+                "transformation produced unexpected value",
+            )),
+        }
+    }
+
+    async fn run_inner(&self) -> Result<()> {
+        let opts = self.opts.clone();
+        // `ClientConfig::create` does blocking I/O.
+        // Same for subscribe, most likely.
+        let consumer = spawn_blocking(move || {
+            let topic = opts.topic.clone();
+
+            let consumer = opts.create_consumer()?;
+            tracing::debug!("Created StreamConsumer");
+
+            consumer.subscribe(&[&topic])?;
+            tracing::debug!(topic, "Subscribed");
+
+            Ok::<_, KafkaError>(consumer)
+        })
+        .await
+        .expect("create_consumer task panicked")?;
+
+        loop {
+            // FIXME(jplatte): I don't know if StreamConsumer::recv has an internal buffer.
+            // Overall, rdkafka seems to be doing a bunch of background magic so maybe it does.
+            // In that case, it's likely already doing batching reads (which don't seem to be
+            // a thing in the public API) internally.
+            // If not, we should likely do some sort of batching ourselves, e.g. have two separate
+            // tokio tasks, one which pulls messages from Kafka and one that processes them, with
+            // a bounded channel in between for backpressure.
+            let msg = consumer.recv().await?;
+            tracing::debug!("Received a message");
+
+            let mut process_error_count = 0;
+            while let Err(e) = self.process(&msg).await {
+                match e {
+                    // If the payload is invalid, log an error and continue.
```

Comment:
I'm a bit confused by how the comments here line up with the code.
> log an error and continue.

and then later

>  break; // don't call process again

FWIW, I also think the code here is relatively easy to follow. The comment `It would fail the same way if retried` seems useful, but I don't think you need the other comments here.